Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMWidevineDecryptorGStreamer.cpp

```diff
@@ -0,0 +1,222 @@
+
+#include "GUniquePtrGStreamer.h"
+#include <gst/gst.h>
+#include <open_cdm.h>
+#include <stdio.h>
+#include <string>
+#include <wtf/glib/GUniquePtr.h>
+#include <wtf/text/WTFString.h>
+
+#define GST_WEBKIT_OPENCDM_WIDEVINE_DECRYPT_GET_PRIVATE(obj) (G_TYPE_INSTANCE_GET_PRIVATE((obj), WEBKIT_OPENCDM_WIDEVINE_TYPE_DECRYPT, WebKitOpenCDMWidevineDecryptPrivate))
+
+struct _WebKitOpenCDMWidevineDecryptPrivate {
+    String m_session;
+    std::unique_ptr<OpenCdm> m_openCdm;
+};
+
+static void webKitMediaOpenCDMDecryptorFinalize(GObject*);
+static gboolean webKitMediaOpenCDMDecryptorHandleKeyResponse(WebKitMediaCommonEncryptionDecrypt*, GstEvent*);
+static gboolean webKitMediaOpenCDMDecryptorDecrypt(WebKitMediaCommonEncryptionDecrypt*, GstBuffer*, GstBuffer*, unsigned, GstBuffer*);
+
+GST_DEBUG_CATEGORY(webkit_media_opencdm_widevine_decrypt_debug_category);
+#define GST_CAT_DEFAULT webkit_media_opencdm_widevine_decrypt_debug_category
+
+#define WIDEVINE_PROTECTION_SYSTEM_UUID "edef8ba9-79d6-4ace-a3c8-27dcd51d21ed"
+
+static GstStaticPadTemplate sinkTemplate = GST_STATIC_PAD_TEMPLATE("sink",
+    GST_PAD_SINK,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS("application/x-cenc, original-media-type=(string)video/webm, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID "; "
+    "application/x-cenc, original-media-type=(string)video/mp4, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID "; "
+    "application/x-cenc, original-media-type=(string)audio/webm, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID "; "
+    "application/x-cenc, original-media-type=(string)audio/mp4, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID "; "
+    "application/x-cenc, original-media-type=(string)video/x-h264, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID "; "
+    "application/x-cenc, original-media-type=(string)audio/mpeg, protection-system=(string)" WIDEVINE_PROTECTION_SYSTEM_UUID ";"));
+
+static GstStaticPadTemplate srcTemplate = GST_STATIC_PAD_TEMPLATE("src",
+    GST_PAD_SRC,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS("video/webm; audio/webm; video/mp4; audio/mp4; audio/mpeg; video/x-h264"));
+
+#define webkit_media_opencdm_widevine_decrypt_parent_class parent_class
+G_DEFINE_TYPE(WebKitOpenCDMWidevineDecrypt, webkit_media_opencdm_widevine_decrypt, WEBKIT_TYPE_MEDIA_CENC_DECRYPT);
+
+static void webkit_media_opencdm_widevine_decrypt_class_init(WebKitOpenCDMWidevineDecryptClass* klass)
+{
+    GObjectClass* gobjectClass = G_OBJECT_CLASS(klass);
+    gobjectClass->finalize = webKitMediaOpenCDMDecryptorFinalize;
+
+    GstElementClass* elementClass = GST_ELEMENT_CLASS(klass);
+    gst_element_class_add_pad_template(elementClass, gst_static_pad_template_get(&sinkTemplate));
+    gst_element_class_add_pad_template(elementClass, gst_static_pad_template_get(&srcTemplate));
+
+    gst_element_class_set_static_metadata(elementClass,
+        "Decrypt content encrypted using OpenCMDi Encryption",
```

Comment:
It has probably been adapted from "PlayReady Encryption", but "OpenCMDi Encryption" does not mean much here. Also there is a typo (CMD instead of CDM), and it should not CDM instead of CDMi.
Maybe:
        "Decrypt content encrypted using a DRM system supported by OpenCDM",